backend/routes/users.js

```diff
@@ -0,0 +1,121 @@
+const express = require("express");
+const router = express.Router();
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const keys = require("../config/keys");
+const passport = require("passport");
+
+// Load input validation
+const validateRegisterInput = require("../validation/signUp");
+const validateLoginInput = require("../validation/signIn");
+
+// Load User model
+const User = require("../models/user");
+
+const axios = require("axios");
+
+router.post("/insertOne", (req, res) => {
+  // Form validation
+  console.log("Received signup request");
+  
+  const { errors, isValid } = validateRegisterInput(req.body.document);
+
+  // Check validation
+  if (!isValid) {
+    return res.status(400).json(errors);
+  }
+  
+  User.findOne({ email: req.body.document.email }).then(user => {
+    if (user) {
+      return res.status(400).json({ email: "Email already exists" });
+    } else {
+      const newUser = new User({
+        name: req.body.document.name,
+        email: req.body.document.email,
+        password: req.body.document.password
+      });
+      
+      // Hash password before saving in database
+      bcrypt.genSalt(10, (err, salt) => {
```

Comment:
the `err` value here is never checked, is that a hole in logic?

Reply:
I will check this asap and see if it is a hole in the logic.